python/pyspark/sql/connect/functions.py

```diff
@@ -92,8 +92,19 @@ def col(col: str) -> Column:
 def lit(col: Any) -> Column:
```

Comment:
actually, PySpark's [`lit` function](https://github.com/apache/spark/blob/master/python/pyspark/sql/functions.py#L168-L179) do not support `tuple` and `dict`, and it just start to support `list` since 3.4.0.

```
In [6]: from pyspark.sql import functions as SF

In [7]: SF.lit({1 : 2})
---------------------------------------------------------------------------
Py4JJavaError                             Traceback (most recent call last)
Cell In[7], line 1
----> 1 SF.lit({1 : 2})
...

Py4JJavaError: An error occurred while calling z:org.apache.spark.sql.functions.lit.
: org.apache.spark.SparkRuntimeException: [UNSUPPORTED_FEATURE.LITERAL_TYPE] The feature is not supported: Literal for '{1=2}' of class java.util.HashMap.
        at 
...

In [8]: SF.lit(("a", "b"))
---------------------------------------------------------------------------
Py4JJavaError                             Traceback (most recent call last)
Cell In[8], line 1
----> 1 SF.lit(("a", "b"))
...

Py4JJavaError: An error occurred while calling z:org.apache.spark.sql.functions.lit.
: org.apache.spark.SparkRuntimeException: [UNSUPPORTED_FEATURE.LITERAL_TYPE] The feature is not supported: Literal for '[a, b]' of class java.util.ArrayList.

...

In [9]: SF.lit(["a", "b"])
Out[9]: Column<'array(a, b)'>
```

I'm thinking of removing `struct` and `map` here.

Reply:
+1

Reply:
Ok, I will send a followup to remove them soon